core/src/Kokkos_Core_fwd.hpp

```diff
@@ -184,6 +186,21 @@ using DefaultHostExecutionSpace KOKKOS_IMPL_DEFAULT_HOST_EXEC_SPACE_ANNOTATION =
     "At least one of the following execution spaces must be defined in order to use Kokkos: Kokkos::OpenMP, Kokkos::Threads, Kokkos::Experimental::HPX, or Kokkos::Serial."
 #endif
 
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP) || \
+    defined(KOKKOS_ENABLE_SYCL)
+#if defined(KOKKOS_ENABLE_CUDA)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION = CudaUVMSpace;
+#elif defined(KOKKOS_ENABLE_HIP)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION = HIPManagedSpace;
+#elif defined(KOKKOS_ENABLE_SYCL)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION =
+    Experimental::SYCLSharedUSMSpace;
+#endif
+#define KOKKOS_HAS_SHARED_SPACE true
+#else
+#define KOKKOS_HAS_SHARED_SPACE false
```

Comment:
No only define it when it's available and set it to `1`
Because our users typically do `#ifdef FOO`